### PR TITLE
feat: deprecate GuildEmbed methods and properties in favour of GuildWidget

### PR DESCRIPTION
**Pull request by NotSugden**
**Please describe the changes this PR makes and why it should be merged:**

This PR deprecates all of the methods and properties relating to `GuildEmbed` as its deprecated from the API (see https://github.com/discord/discord-api-docs/pull/1536), and adds in new methods `fetchWidget` and `setWidget`

I wasn't sure whether to change the existing methods to use the `widget` endpoint since the methods are deprecated, the new methods should instead be used (although it will eventually lead to the old methods breaking)

I also thought it better to rename the existing typedefs (as that's not breaking since the return types are still the same) instead of making duplicate typedefs

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by advaith1** [comment]
My relevant upstream API docs PR has been merged.

File: src/structures/Guild.js [hunk 1]
```diff
@@ -203,6 +204,7 @@ class Guild extends Base {
 
     /**
      * Whether embedded images are enabled on this guild
+     * @deprecated
      * @type {boolean}
```
Comment (SpaceEEC) [inline src/structures/Guild.js:208]:
```suggestion
     * @type {boolean}
     * @deprecated
```

File: src/structures/Guild.js [hunk 2]
```diff
@@ -247,6 +249,7 @@ class Guild extends Base {
 
     /**
      * The embed channel ID, if enabled
+     * @deprecated
      * @type {?string}
      * @name Guild#embedChannelID
```
Comment (SpaceEEC) [inline src/structures/Guild.js:254]:
```suggestion
     * @type {?string}
     * @name Guild#embedChannelID
     * @deprecated
```

File: src/structures/Guild.js [hunk 3]
```diff
@@ -524,6 +527,7 @@ class Guild extends Base {
 
   /**
    * Embed channel for this guild
+   * @deprecated
    * @type {?TextChannel}
    * @readonly
```
Comment (SpaceEEC) [inline src/structures/Guild.js:532]:
```suggestion
   * @type {?TextChannel}
   * @readonly
   * @deprecated
```

File: src/structures/Guild.js [hunk 4]
```diff
@@ -1207,7 +1238,8 @@ class Guild extends Base {
 
   /**
    * Edits the guild's embed.
-   * @param {GuildEmbedData} embed The embed for the guild
+   * @deprecated
+   * @param {GuildWidgetData} embed The embed for the guild
    * @param {string} [reason] Reason for changing the guild's embed
    * @returns {Promise<Guild>}
```
Comment (SpaceEEC) [inline src/structures/Guild.js:1244]:
```suggestion
   * @param {GuildWidgetData} embed The embed for the guild
   * @param {string} [reason] Reason for changing the guild's embed
   * @returns {Promise<Guild>}
   * @deprecated
```

Comment (NotSugden) [reply src/structures/Guild.js]: done

File: src/structures/Guild.js [hunk 5]
```diff
@@ -1261,10 +1291,10 @@ class Guild extends Base {
   }
 
   /**
-   * Edits the guild's embed.
```
Comment (SpaceEEC) [inline src/structures/Guild.js:1264]: This line should not be removed 👀

Comment (NotSugden) [reply src/structures/Guild.js:1264]: oops

**Comment by advaith1** [comment]
The deprecated guild embed properties and endpoints are removed in API v8.

File: typings/index.d.ts [hunk 6]
```diff
@@ -2510,9 +2517,9 @@ declare module 'discord.js' {
     banner?: Base64Resolvable;
   }
 
-  interface GuildEmbedData {
+  interface GuildWidgetData {
     enabled: boolean;
-    channel: GuildChannelResolvable | null;
+    channel?: GuildChannelResolvable;
```
Comment (vladfrangu) [inline typings/index.d.ts:2522]: Why is this marked as undefined? It can still be set to null (to mark the widget shouldn't generate an invite)

Comment (NotSugden) [reply typings/index.d.ts:2522]: oops

**Comment by advaith1** [comment]
just realized, would it be a good idea to document the replacements in the descriptions?

**Comment by iCrawl** [comment]
We did not do that in the past.

**Pull request merged**